plugins/osx/osx.plugin.zsh

```diff
@@ -161,6 +161,15 @@ function itunes() {
 		vol)
 			opt="set sound volume to $1" #$1 Due to the shift
 			;;
+		playing)
+			local state=`osascript -e 'tell application "iTunes" to player state as string'`
+			if [ "$state" = "playing" ]; then
+				currenttrack=`osascript -e 'tell application "iTunes" to name of current track as string'`
+				currentartist=`osascript -e 'tell application "iTunes" to artist of current track as string'`
+				echo "Listening to %F{yellow}$currenttrack%f by %F{yellow}$currentartist%f";
```

Comment:
Thanks @mcornella and @apjanke for your help, made some of the changes suggested.